openquake/hazardlib/gsim/heresi_2018.py

```diff
@@ -0,0 +1,1051 @@
+        SDI
+    ])
+
+    #: Supported intensity measure component is random horizontal
+    #: :attr:`~openquake.hazardlib.const.IMC.RANDOM_HORIZONTAL`,
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.RANDOM_HORIZONTAL
+
+    #: Supported standard deviation types are inter-event, intra-event
+    #: and total
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL,
+        const.StdDev.INTER_EVENT,
+        const.StdDev.INTRA_EVENT
+    ])
+
+    #: No sites parameters are required
+    REQUIRES_SITES_PARAMETERS = set()
+
+    #: Required rupture parameters are magnitude
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag',))
+
+    #: Required distance measure is Rjb
+    REQUIRES_DISTANCES = set(('rjb',))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        # extracting dictionary of coefficients specific to required
+        # intensity measure type.
+        imt_per = imt.period
+
+        CoeffsTable_thisT = {
+            0.05: self.Coeffs_0p05,
+            0.10: self.Coeffs_0p10,
+            0.15: self.Coeffs_0p15,
+            0.20: self.Coeffs_0p20,
+            0.25: self.Coeffs_0p25,
+            0.30: self.Coeffs_0p30,
+            0.35: self.Coeffs_0p35,
+            0.40: self.Coeffs_0p40,
+            0.45: self.Coeffs_0p45,
+            0.50: self.Coeffs_0p50,
+            0.55: self.Coeffs_0p55,
+            0.60: self.Coeffs_0p60,
+            0.65: self.Coeffs_0p65,
+            0.70: self.Coeffs_0p70,
+            0.75: self.Coeffs_0p75,
+            0.80: self.Coeffs_0p80,
+            0.85: self.Coeffs_0p85,
+            0.90: self.Coeffs_0p90,
+            0.95: self.Coeffs_0p95,
+            1.00: self.Coeffs_1p00,
+            1.10: self.Coeffs_1p10,
+            1.20: self.Coeffs_1p20,
+            1.30: self.Coeffs_1p30,
+            1.40: self.Coeffs_1p40,
+            1.50: self.Coeffs_1p50,
+            1.60: self.Coeffs_1p60,
+            1.70: self.Coeffs_1p70,
+            1.80: self.Coeffs_1p80,
+            1.90: self.Coeffs_1p90,
+            2.00: self.Coeffs_2p00,
+            2.20: self.Coeffs_2p20,
+            2.40: self.Coeffs_2p40,
+            2.60: self.Coeffs_2p60,
+            2.80: self.Coeffs_2p80,
+            3.00: self.Coeffs_3p00,
+            3.20: self.Coeffs_3p20,
+            3.40: self.Coeffs_3p40,
+            3.60: self.Coeffs_3p60,
+            3.80: self.Coeffs_3p80,
+            4.00: self.Coeffs_4p00,
+            4.20: self.Coeffs_4p20,
+            4.40: self.Coeffs_4p40,
+            4.60: self.Coeffs_4p60,
+            4.80: self.Coeffs_4p80,
+            5.00: self.Coeffs_5p00
+        }.get(imt_per, 0)
+
+        if CoeffsTable_thisT == 0:
+            print('Warning: Period not found')
+            mean = 0
```

Comment:
Is there any reason to raise a warning here rather than an error? These outputs will simply break things further downstream inside a PSHA calculation, probably misdirecting users to the cause of the error. If there are secondary tools that use this GMPE probably it would be safer to raise the error here and place the call to the GMPE inside a try-except block.

Reply:
You're right. This is something that I did at the beginning and never fixed back to raise an error. I'll fix it